src/components/scenes/CreateWalletSelectFiatScene.tsx

```diff
@@ -1,161 +1,225 @@
+import { useHandler } from '../../hooks/useHandler'
+import { useWatch } from '../../hooks/useWatch'
 import s from '../../locales/strings'
 import { getDefaultFiat } from '../../selectors/SettingsSelectors'
-import { connect } from '../../types/reactRedux'
+import { useDispatch, useSelector } from '../../types/reactRedux'
 import { NavigationProp, RouteProp } from '../../types/routerTypes'
-import { FlatListItem, GuiFiatType } from '../../types/types'
+import { GuiFiatType } from '../../types/types'
+import { getWalletName } from '../../util/CurrencyWalletHelpers'
 import { getSupportedFiats } from '../../util/utils'
 import { SceneWrapper } from '../common/SceneWrapper'
-import { cacheStyles, Theme, ThemeProps, withTheme } from '../services/ThemeContext'
-import { OutlinedTextInput } from '../themed/OutlinedTextInput'
+import { ButtonsModal } from '../modals/ButtonsModal'
+import { FiatListModal } from '../modals/FiatListModal'
+import { TextInputModal } from '../modals/TextInputModal'
+import { Airship } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { CreateWalletSelectCryptoRow } from '../themed/CreateWalletSelectCryptoRow'
+import { EdgeText } from '../themed/EdgeText'
+import { MainButton } from '../themed/MainButton'
 import { SceneHeader } from '../themed/SceneHeader'
 import { SelectableRow } from '../themed/SelectableRow'
+import { WalletCreateItem } from '../themed/WalletList'
 
 type OwnProps = {
   navigation: NavigationProp<'createWalletSelectFiat'>
   route: RouteProp<'createWalletSelectFiat'>
 }
-type StateProps = {
-  supportedFiats: GuiFiatType[]
-}
-type Props = OwnProps & StateProps & ThemeProps
 
-type State = {
-  searchTerm: string
-  selectedFiat: string
-}
+type Props = OwnProps
 
-export class CreateWalletSelectFiatComponent extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      searchTerm: '',
-      selectedFiat: ''
-    }
-  }
+const CreateWalletSelectFiatComponent = (props: Props) => {
+  const { navigation, route } = props
+  const { createWalletList } = route.params
 
-  isValidFiatType = () => {
-    const { selectedFiat } = this.state
-    const fiatTypeIndex = this.props.supportedFiats.findIndex(fiatType => fiatType.value === selectedFiat)
-    const isValid = fiatTypeIndex >= 0
-    return isValid
-  }
+  const dispatch = useDispatch()
+  const theme = useTheme()
+  const styles = getStyles(theme)
 
-  getFiatType = (fiatKey: string) => {
-    const fiatTypeIndex = this.props.supportedFiats.findIndex(fiatType => fiatType.value === fiatKey)
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
 
-    return this.props.supportedFiats[fiatTypeIndex]
-  }
+  const defaultFiat = useSelector(state => getDefaultFiat(state))
+  const [fiat, setFiat] = React.useState(getSupportedFiats(defaultFiat)[0])
```

Comment:
The `getSupportedFiats` call will run on every render, which is expensive. If you put the `useState` initializer in a function, it will only run on first render. So `React.useState(() => getSupportedFiats(defaultFiat)[0])`